Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -255,53 +268,56 @@ class DatabaseContainer: NSPersistentContainer {
     /// Tries to load a persistent store.
     ///
     /// If it fails, for example because of non-matching models, it removes the store, recreates is, and tries to load it again.
-    /// If the second loading fails, too, it throws an error.
     ///
-    private func setupPersistentStore() throws {
-        var storeLoadingError: Error?
-        
-        loadPersistentStores { _, error in
-            storeLoadingError = error
-        }
-        
-        if storeLoadingError != nil {
-            try recreatePersistentStore()
+    private func setupPersistentStore(completion: ((Error?) -> Void)? = nil) {
+        loadPersistentStores { [weak self] _, error in
+            if error != nil {
+                do {
+                    try self?.recreatePersistentStore(completion: completion)
+                } catch {
+                    fatalError("Failed to initialize the storage with error: \(error). This is a non-recoverable error.")
+                }
+            }
         }
     }
     
     /// Removes the loaded persistent store and tries to recreate it.
-    func recreatePersistentStore() throws {
+    func recreatePersistentStore(completion: ((Error?) -> Void)? = nil) throws {
```

Comment:
nit: now, when completion with input `error` is in place, it's a bit weird the func throws, right? 🤔 but not sure if it's smth we wanna fix

Reply:
It actually makes sense: if the error is sync, we can act on it right away (switch to in-memory DB). if the error is async, we can't really do much